routes/project/location/postgres.rb

```diff
@@ -40,6 +40,33 @@ class Clover
         204
       end
 
+      r.patch do
+        authorize("Postgres:edit", pg.id)
+
+        allowed_optional_parameters = ["size", "storage_size", "ha_type"]
+        ignored_parameters = ["family"]
+        request_body_params = validate_request_params([], allowed_optional_parameters, ignored_parameters)
+
+        parsed_target_vm_size = Validation.validate_postgres_size(pg.location, request_body_params["size"] || pg.target_vm_size)
+        target_storage_size_gib = Validation.validate_postgres_storage_size(pg.location, parsed_target_vm_size.vm_size, request_body_params["storage_size"] || pg.target_storage_size_gib)
+        ha_type = request_body_params["ha_type"] || PostgresResource::HaType::NONE
+        Validation.validate_postgres_ha_type(ha_type)
+
+        requested_standby_count = PostgresResource::TARGET_STANDBY_COUNT_MAP[ha_type]
+        requested_postgres_vcpu_count = (requested_standby_count + 1) * parsed_target_vm_size.vcpu
+        Validation.validate_vcpu_quota(@project, "PostgresVCpu", requested_postgres_vcpu_count)
+
+        DB.transaction do
+          pg.update(
+            target_vm_size: parsed_target_vm_size.vm_size,
+            target_storage_size_gib: target_storage_size_gib,
```

Comment:
Should we check the disk usage and not allow downgrading below that? If the disk is already full, this operation will fail anyway.